openslides/core/static/js/core/base.js

```diff
@@ -493,22 +499,46 @@ angular.module('OpenSlidesApp.core', [
 
 .directive('templateHook', [
     '$compile',
+    '$http',
+    '$q',
     'templateHooks',
-    function ($compile, templateHooks) {
+    function ($compile, $http, $q, templateHooks) {
         return {
             restrict: 'E',
             template: '',
             link: function (scope, iElement, iAttr) {
                 var hooks = templateHooks.hooks[iAttr.hookName];
-                var html;
                 if (hooks) {
-                    html = hooks.map(function (hook) {
-                        return '<div>' + hook.template + '</div>';
-                    }).join('');
-                } else {
-                    html = '';
+                    var promises = _.map(hooks, function (hook) {
+                        // Either a template (html given as string) or a templateUrl has
+                        // to be given. If a scope is provided, the schope of this templateHook
+                        // is populated with the given functions/values.
+                        if (hook.template) {
+                            return $q(function (resolve) {
+                                resolve('<div>' + hook.template + '</div>');
+                            });
+                        } else {
+                            return $q(function (resolve) {
+                                $http.get(hook.templateUrl).then(function (success) {
+                                    resolve(success.data);
+                                });
+                            });
+                        }
+                    });
+                    // Populate scope
+                    _.forEach(hooks, function (hook) {
```

Comment:
I wonder whether it is useful to do the scope stuff inside the first "loop" with the templates.